### ISSUE TITLE
Line breaks are not preserved in Notes

### ISSUE THREAD

**Issue by rajatarora** [hadynz/obsidian-kindle-plugin#67]
Whenever there are line breaks in Kindle notes, they are not preserved when those notes are synced to Obsidian. For example, in the following screenshot from https://read.amazon.com/notebook --

![image](https://user-images.githubusercontent.com/318895/124122211-77ce7980-da93-11eb-9937-87f763338598.png)

I have a line break between two paragraphs. The corresponding HTML code also has two `<br/>` tags:

```html
<span id="note" class="a-size-base-plus a-color-base">
  Story about firefighters in Montana. They had jumped off a plane into a burning forest to try and contain the fire. But they quickly found out that they couldn't handle it and will have to run to safety. One of the firefighters started another fire to clear away the fuel in front of him, and lay down on the ground. 
  <br>
  <br>
  The fire reached him, bntouldn't burn him because the ground around him provided no fuel to fire. He survived. 12 others didn't.
</span>
```
However, when this note is synced to Obsidian, it loses the line breaks (the screenshot is from Edit mode):

![image](https://user-images.githubusercontent.com/318895/124122800-1f4bac00-da94-11eb-9ceb-946056fbdd8e.png)

I also looked at the code used to parse the notes, but couldn't find anything wrong with it at first glance:

https://github.com/hadynz/obsidian-kindle-plugin/blob/5ad275fa8d14bcacf0cdc2347b3a549a11e15356/src/scraper/scrapeBookHighlights.ts#L57

Take a look into this please? It is mildly annoying!

**Comment by hadynz** [comment]
Thanks for raising this issue @rajatarora. It is quite a significant find as preserving line breaks is an important requirement that helps a user understand context of their highlights.

> I also looked at the code used to parse the notes, but couldn't find anything wrong with it at first glance:
The plugin uses [cheerio][2] for parsing HTML. The `cheerio.text()` method strips out all HTML. This includes any break lines meaning that we lose the ability of knowing when break lines used to exist. I would have to come up with a solution that does not use `cheerio.text` as it [doesn't support this functionality][1].

[1]: https://github.com/cheeriojs/cheerio/issues/839
[2]: https://cheerio.js.org/

**Comment by hadynz** [comment]
I've looked into this briefly, and ran into a couple of interesting things.

1. Not a single book highlight in my library are showing any break lines in Amazon's notebook. 
2. I haven't been able to recreate your scenario manually!

For a random book, I highlighted something that spans paragraphs. I did this on the Mac Kindle app:
![image](https://user-images.githubusercontent.com/315585/124379482-727b6400-dd0b-11eb-9345-607f698c9143.png)

After syncing my highlights to the cloud, the very same highlight came through Amazon's notebook stripped of any break lines.
![image](https://user-images.githubusercontent.com/315585/124379493-8626ca80-dd0b-11eb-8703-23f4f5df17da.png)

@rajatarora would be interested to know what versions of Kindle do you use for highlighting. Also, do you have many highlights that are missing break lines, or did you simply run into the one exception?

I would prefer if I can find a way to recreate the problem in order to make a fix and test it with confidence. So keen to understand how to recreate one of your examples.

**Comment by rajatarora** [comment]
@hadynz In my example, it's not a highlight but a note! Here's what I did to reproduce the issue just now:

1. Opened my book on https://read.amazon.com

2. Selected some text and clicked on "Note" on the popout menu: 
![image](https://user-images.githubusercontent.com/318895/124381473-66c67800-dce0-11eb-9d6d-7acc1d31dc9c.png)

3. Wrote the note with line breaks: 
![image](https://user-images.githubusercontent.com/318895/124381532-b311b800-dce0-11eb-9b9b-5a0552a02f78.png)

4. It showed up with line breaks intact on https://read.amazon.com/notebook 
![image](https://user-images.githubusercontent.com/318895/124381584-df2d3900-dce0-11eb-978c-4d22e81706d9.png)

The resulting HTML looks like this:

```html
<span id="note" class="a-size-base-plus a-color-base">
  This is a note on a Kindle highlight. I use this functionality to jot down my own thoughts on a particular passage in the book.
  <br>
  <br>
  Let's save this note and check if the line breaks are preserved in Kindle Notebook or not!
</span>
```

And that's the note synced to Obsidian:
![image](https://user-images.githubusercontent.com/318895/124382088-346a4a00-dce3-11eb-9bd8-60e9b2619722.png)

I was also curious about the scenario you mentioned and I was able to reproduce it too -- a multi-paragraph "highlight" (as opposed to a "note") gets stripped off of its line breaks in the Kindle notebook itself! To tell you the truth, I don't usually highlight large passages in my Kindle books so I did not encounter this particular scenario. This is certainly an issue for Amazon to fix.

And yes, all of my notes with line breaks are missing them when synced to Obsidian. 

I use Kindle on quite a few devices:

1. Linux desktop: Kindle Cloud Reader on Firefox 89
2. Macbook Pro: Kindle Mac App v 1.31.0
3. iPad Pro: Kindle for iPad App v6..43
4. Kindle Voyage: Firmware version 5.13.6 (3731990038)

I hope this information will help you reproduce the issue at your end. If not, I'll be happy to answer more questions :smile:

**Comment by hadynz** [comment]
Ahhh. Gotcha. You've mentioned this is about notes in the issue title and I still completely missed it. 

Cool, thanks for your reply. I should be able to replicate this easily and fix.

**Comment by hadynz** [comment]
This has been fixed in [0.2.18][1]. Please give it a try and let me know if it worked for your highlights. 

[1]: https://github.com/hadynz/obsidian-kindle-plugin/releases/tag/0.2.18

**Comment by rajatarora** [comment]
I used 0.2.18 to re-sync my books to Obsidian. Line breaks are showing up properly now! Thank you :)
